Azure.Core.Pole.Tests/TypeSpecGenerator.cs

```diff
@@ -2,20 +2,62 @@
 // Licensed under the MIT License.
 
 using Azure.Core.Pole.Tooling;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
 using NUnit.Framework;
+using System;
 using System.IO;
+using System.Linq;
+using System.Reflection;
 
 namespace Azure.Core.Pole.Tests
 {
     public partial class TypeSpecGeneratorTests
     {
         [Test]
-        public void Basics()
+        public void DogModelGeneration()
```

Comment:
Great. We now have a test that generates source code based on tsp. Let's now add a separate test that similarly generates the source code, but then compiles it, instantiates Doc class, sets its properties, reads them and verifies that the Dog class actually works